template/Makefile.j2

```diff
@@ -59,7 +59,18 @@ docker-publish:
 	fi;\
 	# This generates a signature and publishes it to the registry, next to the image\
 	# Uses the keyless signing flow with Github Actions as identity provider\
-	cosign sign -y ${OCI_REGISTRY_HOSTNAME}/${OCI_REGISTRY_PROJECT_IMAGES}/${OPERATOR_NAME}:@$$REPO_DIGEST_OF_IMAGE
+	cosign sign -y ${OCI_REGISTRY_HOSTNAME}/${OCI_REGISTRY_PROJECT_IMAGES}/${OPERATOR_NAME}@$$REPO_DIGEST_OF_IMAGE;\
+	# Generate the SBOM for the operator image, this leverages the already generated SBOM for the operator binary by cargo-cyclonedx\
+	syft scan -o cyclonedx-json --exclude "/usr/local/bin/stackable-${OPERATOR_NAME}" --scope all-layers --source-name "${OPERATOR_NAME}" --source-version "${VERSION}" ${OCI_REGISTRY_HOSTNAME}/${OCI_REGISTRY_PROJECT_IMAGES}/${OPERATOR_NAME}@$$REPO_DIGEST_OF_IMAGE > sbom.json;\
```

Comment:
- syft also has a -f option to write to a file directly
- I believe syft can "pretty print" output, not sure if that's the default, if it's not I think we should enable it
- With syft you can enable/disable catalogers, maybe we need to disable the cataloger responsible for finding cargo-auditable things to avoid the duplicates

Reply:
- I added the flag `--output cyclonedx-json=sbom.json` (`--file` is deprecated)
 - By default, Syft spits out compact JSON, which is fine I think, since the SBOM is saved base64 encoded in the OCI registry, so customers need to do some decoding anyway, they don't just download a JSON file. I think how customers see it depends on which tool they use to view the decoded JSON (could also be some GUI / SBOM visualizer). So I'd say: Save it as compact JSON (for efficiency reasons), pretty printing is responsibility of the client. Not a strong opinion though.
 - Sounds good, I'll take a look at the catalogers later

Reply:
Disabled the "cargo audit binary" cataloger now, tested it in Linux and MacOS, works as expected. Testrun with airflow-operator was also successful (https://github.com/stackabletech/airflow-operator/actions/runs/7665989842).